staging/src/k8s.io/csi-translation-lib/translate.go

```diff
@@ -35,9 +35,36 @@ var (
 	}
 )
 
+// CSITranslator translates in-tree storage API objects to their equivilant CSI
+// API objects. It also provides many helper functions to determine whether
+// translation logic exists and the mappings between "in-tree plugin <-> csi driver"
+type CSITranslator interface {
+	TranslateInTreeStorageClassToCSI(inTreePluginName string, sc *storage.StorageClass) (*storage.StorageClass, error)
+	TranslateInTreeInlineVolumeToCSI(volume *v1.Volume) (*v1.PersistentVolume, error)
+	TranslateInTreePVToCSI(pv *v1.PersistentVolume) (*v1.PersistentVolume, error)
+	TranslateCSIPVToInTree(pv *v1.PersistentVolume) (*v1.PersistentVolume, error)
+	IsMigratableIntreePluginByName(inTreePluginName string) bool
+	IsMigratedCSIDriverByName(csiPluginName string) bool
+	GetInTreePluginNameFromSpec(pv *v1.PersistentVolume, vol *v1.Volume) (string, error)
+	GetCSINameFromInTreeName(pluginName string) (string, error)
+	GetInTreeNameFromCSIName(pluginName string) (string, error)
+	IsPVMigratable(pv *v1.PersistentVolume) bool
+	IsInlineMigratable(vol *v1.Volume) bool
+}
+
+type csiTranslator struct{}
+
+var _ CSITranslator = &csiTranslator{}
+
+// NewCSITranslator returns the csiTranslator struct which does real translation
+// for "in-tree plugins <-> csi drivers"
+func NewCSITranslator() CSITranslator {
```

Comment:
If the only "thing" in this package is the translator and the only "constructor" is this func, then you can get away with calling this `New()`. It will be unambiguous what `csitrans.New()` means. 

(It will show up at the right place in godoc, regardless of the function name.)